llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -613,7 +627,37 @@ GCNScheduleDAGMILive::getBBLiveInMap() const {
       ++I;
     } while (I != E && I->first->getParent() == BB);
   } while (I != E);
-  return getLiveRegMap(BBStarters, false /*After*/, *LIS);
+  return getLiveRegMap(BBStarters, /*After=*/false, *LIS);
+}
+
+DenseMap<MachineInstr *, GCNRPTracker::LiveRegSet>
+GCNScheduleDAGMILive::getBBLiveOutMap() const {
+  assert(!Regions.empty());
+  std::vector<MachineInstr *> BBEnders;
+  BBEnders.reserve(Regions.size());
+  auto I = Regions.rbegin(), E = Regions.rend();
+  for (; I != E; I++)
+    BBEnders.push_back(getLastMIForRegion(I->first, I->second));
+
+  return getLiveRegMap(BBEnders, /*After= */ true, *LIS);
+}
+
+void RegionPressureMap::buildLiveRegMap() {
+  if (IsMapGenerated) {
+    IdxToInstruction.clear();
+    BBLiveRegMap.clear();
+    IsMapGenerated = false;
+  }
+
+  BBLiveRegMap = IsLiveOut ? DAG->getBBLiveOutMap() : DAG->getBBLiveInMap();
+  for (unsigned I = 0; I < DAG->Regions.size(); I++) {
+    MachineInstr *RegionKey;
+    RegionKey = IsLiveOut ? getLastMIForRegion(DAG->Regions[I].first,
+                                               DAG->Regions[I].second)
+                          : &*DAG->Regions[I].first;
+    IdxToInstruction[I] = RegionKey;
+  }
+  IsMapGenerated = true;
```

Comment:
Don't understand why this needs to be tracked, this is the same as the map is empty or not?